pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -150,11 +150,32 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 				glog.V(3).Infof("Considering Task <%v/%v> on node <%v>. Task request: <%v>; Idle: <%v>; Used: <%v>; Releasing: <%v>; Backfilled: <%v>",
 					task.Namespace, task.Name, node.Name, task.Resreq, node.Idle, node.Used, node.Releasing, node.Backfilled)
 
-				if task.InitResreq.LessEqual(node.GetAccessibleResource()) {
-					glog.V(3).Infof("Binding Task <%v/%v> to node <%v>",
-						task.Namespace, task.Name, node.Name)
+				// allow over-allocate for starving job 
+				isJobStarving := job.Starving(ssn.StarvationThreshold)
+				toOverAllocateForStarv := false
+				if isJobStarving {
+					netResource := node.Allocatable.Clone()
+					for _, nodeTask := range node.Tasks {
+						if nodeTask.Job == job.UID &&
```

Comment:
This if condition is pretty complicated. It will be great if you can simplify it. 

I do not understand this if condition. I thought we would like to remove resources occupied by non kube-batch pods from net resource. Is my understanding correct?

Reply:
fixed.